tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -18,9 +18,28 @@ namespace Datadog.Trace.AppSec.Waf
 {
     internal static class Encoder
     {
+        private const int MaxBytesForMaxStringLength = WafConstants.MaxStringLength * 4;
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(Encoder));
         private static readonly int ObjectStructSize = Marshal.SizeOf(typeof(DdwafObjectStruct));
 
+        [ThreadStatic]
```

Comment:
for my understanding, why is it necessary? 😅

Reply:
The Unmanaged Pool is not thread safe for performance reasons, so the best way to avoid leaking memory in a concurrent scenario like ASP.NET is to create and mantain a pool for each thread (and just dispose the pool if the thread get's killed), this pool instance is copied to the Encoder stack so we avoid accessing a thread static twice in the same processing.